### PR TITLE
executor: range task channel in XSelectIndexExec.Close.

### PR DESCRIPTION
**Pull request by coocood**
Fixes goroutine leak.

### CODE REVIEW

**Comment by coocood** [comment]
@shenli @tiancaiamao PTAL

File: executor/distsql.go [hunk 1]
```diff
@@ -505,11 +517,6 @@ func (e *XSelectIndexExec) nextForDoubleRead() (*Row, error) {
 		}
 		row, err := e.taskCurr.getRow()
 		if err != nil {
-			// Consume the task channel in case channel is full.
```
Comment (coocood) [inline executor/distsql.go]: It is moved to Close method.

File: executor/distsql_test.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	"github.com/pingcap/tidb/executor"
+	"github.com/pingcap/tidb/store/tikv"
+	"github.com/pingcap/tidb/util/testkit"
+)
+
+// This test checks that when a index double read returns before reading all the rows, the goroutine doesn't
+// leak. For testing distsql with multiple regions, we need to manually split a mock TiKV.
+func (s *testSuite) TestIndexDoubleReadClose(c *C) {
+	if _, ok := s.store.GetClient().(*tikv.CopClient); !ok {
+		// Make sure the store is tikv store.
```
Comment (shenli) [inline executor/distsql_test.go]: Should we assert ok is True here? Or the test case may be ignored.

Comment (coocood) [inline executor/distsql_test.go]: The test can be configured to run with localstore, then the client is not `*tikv.CopClient`.

**Comment by shenli** [comment]
LGTM

**Comment by shenli** [comment]
@tiancaiamao @zimulala PTAL

**Comment by zimulala** [comment]
LGTM

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**